src/Two/AbstractProvider.php

```diff
@@ -314,7 +314,10 @@ public function getAccessTokenResponse($code)
      */
     protected function getTokenHeaders($code)
     {
-        return ['Accept' => 'application/json'];
+        return [
+            'Accept' => 'application/json',
+            'Authorization' => 'Basic '.base64_encode($this->clientId.':'.$this->clientSecret),
```

Comment:
Why this combination of client id or secret? Can you share the spec for this?

Reply:
Hey Dries! Thanks a lot for the super quick reply.

The spec says:

> Clients in possession of a client password MAY use the HTTP Basic
> authentication scheme as defined in [[RFC2617](https://datatracker.ietf.org/doc/html/rfc2617)] to authenticate with
>the authorization server.  The client identifier is encoded using the
> "application/x-www-form-urlencoded" encoding algorithm per
> [Appendix B](https://datatracker.ietf.org/doc/html/rfc6749#appendix-B), and the encoded value is used as the username; the client
> password is encoded using the same algorithm and used as the
> password.  The authorization server MUST support the HTTP Basic
> authentication scheme for authenticating clients that were issued a
> client password.

So this combination of client id or secret as user name / password is directly mandated by the spec.
After, that, the encoding (the Basic auth is base 64 encoded) is mandated by  RFC2617.

Reply:
Thanks but sorry I don't follow. Why the `id:secret` combo/syntax that's then base64_encoded? I don't read anything about that in the paragraph you shared.

Reply:
No problem: The [RFC2617](https://datatracker.ietf.org/doc/html/rfc2617) explains how to do "Basic authentication". In the [section 2](https://datatracker.ietf.org/doc/html/rfc2617#section-2), it says that the Basic authentication expect a user name and a password, separated by a ":" that is base64 encoded:

> To receive authorization, the client sends the userid and password,
>   separated by a single colon (":") character, within a base64 [[7](https://datatracker.ietf.org/doc/html/rfc2617#ref-7)]
>   encoded string in the credentials.

The [RFC-6749](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1)  simply says that all servers MUST support Basic authentication (as per the RFC2617) and provides the mapping. In the context of RFC-6749:

- the username is actually the client id
- the password is actually the secret

Reply:
Thanks @moufmouf for explaining 👍